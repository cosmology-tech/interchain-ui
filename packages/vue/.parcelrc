{
  "extends": "@parcel/config-default",
  "resolvers": [
    "parcel-resolver-ts-base-url",
    "..."
  ],
  "transformers": {
    "*.{svg,png}": [
      "@parcel/transformer-raw"
    ],
    "*.css.ts": [
      "@vanilla-extract/parcel-transformer",
      "@parcel/transformer-css"
    ],
    "*.{vue}": [
      "@parcel/transformer-vue"
    ]
  },
  "optimizers": {
    "*.js": [
      "parcel-optimizer-unlink-css",
      "@parcel/optimizer-terser"
    ]
  }
}
