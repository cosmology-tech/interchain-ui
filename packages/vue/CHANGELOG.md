# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.7](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.6...@interchain-ui/vue@1.1.7) (2023-07-12)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.6](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.5...@interchain-ui/vue@1.1.6) (2023-07-12)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.5](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.4...@interchain-ui/vue@1.1.5) (2023-07-10)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.4](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.3...@interchain-ui/vue@1.1.4) (2023-07-10)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.3](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.2...@interchain-ui/vue@1.1.3) (2023-07-10)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.2](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.1...@interchain-ui/vue@1.1.2) (2023-07-10)

**Note:** Version bump only for package @interchain-ui/vue

## [1.1.1](https://github.com/cosmology-tech/interchain-ui/compare/@interchain-ui/vue@1.1.0...@interchain-ui/vue@1.1.1) (2023-07-09)

**Note:** Version bump only for package @interchain-ui/vue

# 1.1.0 (2023-07-09)

### Bug Fixes

- improve and fix some nextjs related bugs ([#32](https://github.com/cosmology-tech/interchain-ui/issues/32)) ([ae5b352](https://github.com/cosmology-tech/interchain-ui/commit/ae5b35247b1a2e12f956363761c56e00e1fb9818))
- layout anim and update deps ([#39](https://github.com/cosmology-tech/interchain-ui/issues/39)) ([55f1af7](https://github.com/cosmology-tech/interchain-ui/commit/55f1af7fb7428de1095863f8a79287859a629b6f))
- optimize dev script ([52b7e48](https://github.com/cosmology-tech/interchain-ui/commit/52b7e483579161bedce97ece68090c0a9b58e54c))
- temporarily remove layout animation ([f14ad8b](https://github.com/cosmology-tech/interchain-ui/commit/f14ad8b4702cb01c41d08988219dfc93f26ab8ae))
- TS errors ([#36](https://github.com/cosmology-tech/interchain-ui/issues/36)) ([85d4732](https://github.com/cosmology-tech/interchain-ui/commit/85d4732d735620416007084a345bc01ba8f42fd9))

### Features

- base setup ([#2](https://github.com/cosmology-tech/interchain-ui/issues/2)) ([968b250](https://github.com/cosmology-tech/interchain-ui/commit/968b250bbb2a8103ee2973f9f23722df505f0e40))
- connect modal ([#12](https://github.com/cosmology-tech/interchain-ui/issues/12)) ([c59c9d6](https://github.com/cosmology-tech/interchain-ui/commit/c59c9d69317751472d71916cc8608396ae801e33))
- faster build ([#6](https://github.com/cosmology-tech/interchain-ui/issues/6)) ([c59dde7](https://github.com/cosmology-tech/interchain-ui/commit/c59dde76b908f326695928b5cf09c822b51d26a6))
- improve build time ([#4](https://github.com/cosmology-tech/interchain-ui/issues/4)) ([fcd2bba](https://github.com/cosmology-tech/interchain-ui/commit/fcd2bbad4c66242cb7f38a75ae7aa9ade9e5a46d))
- move back to yarn and add icon ([#3](https://github.com/cosmology-tech/interchain-ui/issues/3)) ([8372ff1](https://github.com/cosmology-tech/interchain-ui/commit/8372ff1df66906850512b3948a146048361469f8))
