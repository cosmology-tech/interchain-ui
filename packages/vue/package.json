{
  "name": "@cosmology-mitosis/vue",
  "version": "1.0.0",
  "type": "module",
  "private": true,
  "description": "> TODO: description",
  "author": "yyyyaaa <phatghaa@gmail.com>",
  "homepage": "https://github.com/cosmology-tech/cosmology-mitosis/tree/main/packages/vue#readme",
  "license": "SEE LICENSE IN LICENSE",
  "main": "./dist/cosmology-mitosis-vue.umd.js",
  "module": "./dist/cosmology-mitosis-vue.es.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/cosmology-mitosis-vue.es.js",
      "require": "./dist/cosmology-mitosis-vue.umd.js"
    }
  },
  "dependencies": {
    "@fastify/deepmerge": "^1.3.0",
    "@vanilla-extract/css": "^1.9.5",
    "@vanilla-extract/dynamic": "^2.0.3",
    "@vanilla-extract/recipes": "^0.3.0",
    "@vanilla-extract/sprinkles": "^1.5.1",
    "@zag-js/dialog": "^0.2.13",
    "@zag-js/vue": "^0.2.10",
    "animejs": "^3.2.1",
    "zustand": "^4.3.6",
    "immer": "^9.0.19"
  },
  "devDependencies": {
    "clean-css-cli": "^5.6.2"
  },
  "peerDependencies": {
    "vue": "^3.x"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "registry": "https://registry.npmjs.org"
  },
  "scripts": {
    "prebuild": "yarn install",
    "build": "rollup -c --bundleConfigAsCjs",
    "build:dev": "cross-env NODE_ENV=development rollup -c --bundleConfigAsCjs",
    "clean": "rimraf ./dist",
    "dts": "tsc -p tsconfig.json --emitDeclarationOnly --declaration --declarationDir ./dist --skipLibCheck",
    "minifyCss": "cleancss --batch --batch-suffix '' dist/**/*.css"
  }
}
