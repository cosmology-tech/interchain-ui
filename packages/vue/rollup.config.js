const vue = require("rollup-plugin-vue");
const ts = require("rollup-plugin-typescript");
const config = require("../../rollup.config");

module.exports = config({
  dir: __dirname,
  packageJson: require("./package.json"),
  plugins: [
    ts({
      tsconfig: false,
      experimentalDecorators: true,
      module: "es2015",
    }),
    vue(),
  ],
  external: ["vue"],
  dts: false,
});
