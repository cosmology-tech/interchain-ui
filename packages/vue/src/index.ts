import { setPlatform, Platform } from "./helpers";
export { setDebugLevel, DebugLevel } from "./helpers";
export { store } from "./models/store";

export { default as Button } from "./ui/button/button.vue";
export { default as ThemeProvider } from "./ui/theme-provider/theme-provider.vue";
export { default as WalletConnectModal } from "./ui/wallet-connect-modal/wallet-connect-modal.vue";
export { default as FadeIn } from "./ui/fade-in/fade-in.vue";

setPlatform(Platform.Vue);
