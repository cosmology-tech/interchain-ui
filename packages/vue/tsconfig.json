{
  "compilerOptions": {
    // Enable latest features
    "lib": ["ESNext", "DOM", "DOM.Iterable"],
    "target": "ESNext",
    "module": "ESNext",
    "moduleDetection": "force",
    "allowJs": true,

    // Bundler mode
    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "verbatimModuleSyntax": false,
    "noEmit": true,

    // Best practices
    "strict": true,
    "skipLibCheck": true,
    "noFallthroughCasesInSwitch": true,
    "noPropertyAccessFromIndexSignature": false,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "customConditions": ["source"],
    "types": ["vite/client", "node"],

    // Added from React tsconfig
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true,
    "isolatedModules": true,

    // Path aliases
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    },

    // Vue specific
    "jsx": "preserve",
    "jsxImportSource": "vue"
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "scaffolds/**/*.ts",
    "scaffolds/**/*.d.ts",
    "scaffolds/**/*.tsx",
    "scaffolds/**/*.vue"
  ],
  "exclude": ["node_modules", "dist"]
}
