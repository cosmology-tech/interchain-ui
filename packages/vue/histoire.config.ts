import { HstVue } from "@histoire/plugin-vue";
import path from "node:path";
import vue from "@vitejs/plugin-vue";
import { vanillaExtractPlugin } from "@vanilla-extract/vite-plugin";
import { defineConfig } from "histoire";

const aliases = ["ui", "styles", "models", "helpers"];

export default defineConfig({
  plugins: [HstVue()],
  storyMatch: ["**/*.stories.vue"],
  vite(config, env) {
    let finalConfig = {
      ...config,
      plugins: config.plugins ?? [],
      resolve: config.resolve ?? {},
    };

    finalConfig.plugins.push(vue(), vanillaExtractPlugin());
    finalConfig.resolve.alias = aliases.map((alias) => ({
      find: `@/${alias}`,
      replacement: path.resolve(__dirname, `src/${alias}`),
    }));

    return finalConfig;
  }
  },
  // setupFile: ".storybook/main.ts",
});
