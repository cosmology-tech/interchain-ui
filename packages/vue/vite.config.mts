import { globbySync } from "globby";
import { copyFileSync } from "node:fs";
import path from "node:path";
import { defineConfig } from "vite";
import dts from "vite-plugin-dts";
import vue from "@vitejs/plugin-vue";
import { vanillaExtractPlugin } from "@vanilla-extract/vite-plugin";
import ViteVueComplexTypes from "@vue.ts/complex-types/vite";
import pkg from "./package.json";

const aliases = ["ui", "styles", "models", "helpers"];

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    dts({
      entryRoot: "src",
      staticImport: true,
      exclude: [
        "**/*.stories.*",
        "**/*.test.*",
        "**/tests/*",
        "**/examples/*",
        "**/setup-test.ts",
      ],
      afterBuild: () => {
        globbySync(["dist/**/*.d.ts", "dist/**.d.ts"]).map((file) => {
          copyFileSync(file, file.replace(/\.d\.ts$/, ".d.cts"));
        });
      },
    }),
    ViteVueComplexTypes(),
    vanillaExtractPlugin(),
    vue(),
  ],
  build: {
    outDir: "dist",
    target: "esnext",
    minify: false,
    lib: {
      entry: globbySync("src/**/index.ts"),
      fileName: (format) => (format === "es" ? "index.js" : "index.cjs"),
    },
    rollupOptions: {
      external: [
        ...Object.keys(pkg.dependencies ?? {}),
        ...Object.keys(pkg.peerDependencies ?? {}),
      ],
      output: [
        {
          format: "cjs",
          preserveModules: true,
          preserveModulesRoot: "src",
          exports: "named",
          entryFileNames: "[name].cjs",
        },
        {
          format: "es",
          preserveModules: true,
          preserveModulesRoot: "src",
          exports: "named",
          entryFileNames: "[name].js",
        },
      ],
    },
  },
  resolve: {
    conditions: ["source"],
    alias: [
      ...aliases.map((alias) => ({
        find: `@/${alias}`,
        replacement: path.resolve(__dirname, `src/${alias}`),
      })),
      {
        find: "@ark-ui/vue",
        replacement: "@ark-ui/vue/dist/index.mjs",
      },
    ],
  },
});
