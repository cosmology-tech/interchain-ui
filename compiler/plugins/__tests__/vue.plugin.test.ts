import { fixVueEventHandlers } from "../vue.plugin";
import { buttonEventHandlersFixture } from "../__fixtures__/button-event-handlers";

describe("vue plugin", () => {
  it("should fix vue event handlers", () => {
    const result = fixVueEventHandlers(buttonEventHandlersFixture);
    const vBind = 'v-bind="{ ...spreadAttributes, ...eventHandlers }"';
    const vOn = 'v-on="{ ...eventHandlers }"';

    console.log(result);
    expect(result).not.toContain(vBind);
    expect(result).toContain(vOn);
  });
});
