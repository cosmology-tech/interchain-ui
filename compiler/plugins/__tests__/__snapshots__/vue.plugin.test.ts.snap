// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue plugin should fix vue event handlers 1`] = `
"<template>
  <Box
    :boxRef="buttonRef"
    :class="combinedClassName"
    v-bind="{ ...spreadAttributes, ...eventHandlers }"
    ><Spinner
      :size="iconSize"
      :attributes="{
        display:
          isLoading && spinnerPlacement === 'start' ? 'inline-block' : 'none',
      }"
    ></Spinner
    ><Icon
      :name="leftIcon"
      :size="iconSize"
      :attributes="{
        display: !!leftIcon && !isLoading ? 'inline-block' : 'none',
        marginRight: !$slots.default ? '$0' : '$2',
      }"
    ></Icon>
    <template v-if="!isLoading">
      <slot />
    </template>

    <Icon
      :name="rightIcon"
      :size="iconSize"
      :attributes="{
        display: !!rightIcon && !isLoading ? 'inline-block' : 'none',
        marginLeft: !$slots.default ? '$0' : '$2',
      }"
    ></Icon
    ><Spinner
      :size="iconSize"
      :attributes="{
        display:
          isLoading && spinnerPlacement === 'end' ? 'inline-block' : 'none',
      }"
    ></Spinner
  ></Box>
</template>

<script setup lang="ts">
    
    const eventMappings = {
  "onClick": "click",
  "onDoubleClick": "dblclick",
  "onMouseDown": "mousedown",
  "onMouseUp": "mouseup",
  "onMouseEnter": "mouseenter",
  "onMouseLeave": "mouseleave",
  "onMouseMove": "mousemove",
  "onMouseOver": "mouseover",
  "onMouseOut": "mouseout",
  "onKeyDown": "keydown",
  "onKeyUp": "keyup",
  "onKeyPress": "keypress",
  "onFocus": "focus",
  "onBlur": "blur",
  "onInput": "input",
  "onChange": "change",
  "onSubmit": "submit",
  "onReset": "reset",
  "onScroll": "scroll",
  "onWheel": "wheel",
  "onDragStart": "dragstart",
  "onDrag": "drag",
  "onDragEnd": "dragend",
  "onDragEnter": "dragenter",
  "onDragLeave": "dragleave",
  "onDragOver": "dragover",
  "onDrop": "drop",
  "onTouchStart": "touchstart",
  "onTouchMove": "touchmove",
  "onTouchEnd": "touchend",
  "onTouchCancel": "touchcancel"
};
  
    const emit = defineEmits(["click", "dblclick", "mousedown", "mouseup", "mouseenter", "mouseleave", "mousemove", "mouseover", "mouseout", "keydown", "keyup", "keypress", "focus", "blur", "input", "change", "submit", "reset", "scroll", "wheel", "dragstart", "drag", "dragend", "dragenter", "dragleave", "dragover", "drop", "touchstart", "touchmove", "touchend", "touchcancel"]);
const eventMappings = {
  onClick: "click",
  onDoubleClick: "dblclick",
  onMouseDown: "mousedown",
  onMouseUp: "mouseup",
  onMouseEnter: "mouseenter",
  onMouseLeave: "mouseleave",
  onMouseMove: "mousemove",
  onMouseOver: "mouseover",
  onMouseOut: "mouseout",
  onKeyDown: "keydown",
  onKeyUp: "keyup",
  onKeyPress: "keypress",
  onFocus: "focus",
  onBlur: "blur",
  onInput: "input",
  onChange: "change",
  onSubmit: "submit",
  onReset: "reset",
  onScroll: "scroll",
  onWheel: "wheel",
  onDragStart: "dragstart",
  onDrag: "drag",
  onDragEnd: "dragend",
  onDragEnter: "dragenter",
  onDragLeave: "dragleave",
  onDragOver: "dragover",
  onDrop: "drop",
  onTouchStart: "touchstart",
  onTouchMove: "touchmove",
  onTouchEnd: "touchend",
  onTouchCancel: "touchcancel"
};
const emit = defineEmits(["click", "dblclick", "mousedown", "mouseup", "mouseenter", "mouseleave", "mousemove", "mouseover", "mouseout", "keydown", "keyup", "keypress", "focus", "blur", "input", "change", "submit", "reset", "scroll", "wheel", "dragstart", "drag", "dragend", "dragenter", "dragleave", "dragover", "drop", "touchstart", "touchmove", "touchend", "touchcancel"]);
import { computed, onMounted, onUnmounted, ref } from "vue";
import { assignInlineVars } from "@vanilla-extract/dynamic";
import clx from "clsx";
import Icon from "../icon";
import Box from "../box";
import Spinner from "../spinner";
import { store, UIState } from "../../models/store";
import { recipe, buttonOverrides } from "./button.helper";
import { isDefaultAccent, getAccentHover } from "../../helpers/style";
import { themeVars } from "../../styles/themes.css";
import { fullWidth, fullWidthHeight } from "../shared/shared.css";
import type { UnknownRecord } from "../../helpers/types";
import type { ButtonProps } from "./button.types";
import type { ThemeVariant } from "../../models/system.model";
import type { OverrideStyleManager } from "../../styles/override/override";
import * as styles from "./button.css";
const props = withDefaults(defineProps<ButtonProps>(), {
  intent: "primary",
  size: "md",
  as: "button",
  variant: "solid",
  disabled: undefined,
  isLoading: undefined,
  fluidWidth: undefined,
  fluid: undefined,
  className: undefined,
  attributes: undefined,
  domAttributes: undefined,
  buttonRef: undefined,
  iconSize: undefined,
  spinnerPlacement: "start",
  leftIcon: undefined,
  children: undefined,
  rightIcon: undefined
});
const _overrideManager = ref(null);
const _theme = ref("light");
const _themeAccent = ref(null);
const cleanupRef = ref<() => void>();
onMounted(() => {
  const uiStore = getStoreState();
  _theme.value = uiStore[0];
  _themeAccent.value = uiStore[1];
  _overrideManager.value = uiStore[2];
  cleanupRef.value = store.subscribe((newState, prevState) => {
    _theme.value = newState.theme;
    _themeAccent.value = newState.themeAccent;
    _overrideManager.value = newState.overrideStyleManager;
  });
});
onUnmounted(() => {
  if (typeof cleanupRef.value === "function") {
    cleanupRef.value();
  }
});
const combinedClassName = computed(() => {
  return clx(styles.buttonSize[props.size], recipe({
    as: props.as,
    variant: props.variant,
    intent: props.intent ?? "primary",
    isDisabled: props.disabled || props.isLoading,
    theme: getStoreState().theme
  }), props.fluidWidth ? fullWidth : null, props.fluid ? fullWidthHeight : null, props.class);
});
const spreadAttributes = computed(() => {
  return Object.assign({
    as: props.as
  }, {
    attributes: {
      ...props.attributes,
      disabled: props.disabled,
      // style: state.getVars(),
      ...props.domAttributes
    }
  });
});
const eventHandlers = computed(() => {
  const handlers = {};
  Object.entries(eventMappings).forEach(([propName, eventName]) => {
    if (propName in props) {
      handlers[eventName] = event => {
        const propEventHandler = props[propName];
        if (typeof propEventHandler === "function") {
          propEventHandler(event);
        }
        emit(eventName, event);
      };
    }
  });
  return handlers;
});
function getStoreState() {
  // This seems weird but it's a workaround for one minor bug from mitosis
  // If we have any variables in any function scope that has the same name as the store state, mitosis understands that it's the same variable
  // and will attempt to transform those unwanted/unrelated variables into the ones in the state.<variable>
  // So we need to name these values differently (e.g. _keyA: valueA) or inverse
  return {
    theme: store.getState().theme,
    themeAccent: store.getState().themeAccent,
    overrideStyleManager: store.getState().overrideStyleManager
  };
}
function getVars() {
  const accent = _themeAccent.value;
  const isDefaultAppearance = isDefaultAccent(accent) && accent === "blue";

  // Only allow accent customization for 'primary' Intent
  const isPrimaryIntent = props.intent === "primary";
  return isDefaultAppearance || !isPrimaryIntent ? _overrideManager.value?.applyOverrides(buttonOverrides.name) : assignInlineVars({
    [styles.buttonBgVar]: themeVars.colors.accent,
    [styles.buttonTextColorVar]: themeVars.colors.accentText,
    [styles.buttonHoverBgVar]: getAccentHover(themeVars.colors.accent)
  });
}
    </script>
"
`;
